File: codeforces/round49/InvaseColor.cpp
#include<bits/stdc++.h>
using namespace std;
long long f[505][505],n,k,ans,mod=998244353;
int main() {
    cin>>n>>k;
    f[0][0]=1;
    for(int i=0;i<n;++i) for(int j=0;j<=i;++j) for(int k=1;k+i<=n;++k){
        f[i+k][max(j,k)]=(f[i+k][max(j,k)]+f[i][j])%mod;
        cout << "f[" << i+k << "," << max(j,k) << "]=" << f[i+k][max(j,k)] << " f[" << i << ","<<j << "]=" << f[i][j] << endl;
    }
    for(int i=1;i<=n;++i) for(int j=1;j<=n;++j) if(i*j<k) ans=(ans+f[n][i]*f[n][j])%mod;
    cout<<ans*2%mod<<endl;
    return 0;
}

File: codeforces/round49/InvaseColor.py
MOD = 998244353

(n,k) = map(int,input().split())
dp = [[0 for j in range(0,n+5)] for i in range(0,n+5)]

result = 0
dp[0][0] = 1
for i in range(0,n):
    for j in range(0,i+1):
        for t in range(i+1,n+1):
            dp[t][max(j,t-i)] = (dp[t][max(j,t-i)] + dp[i][j])%MOD

for i in range(1,n+1):
    for j in range(1,n+1):
        if i*j < k:
            result = (result + dp[n][i]*dp[n][j]) % MOD
print((result*2) % MOD)

